src/oemof/solph/flows/_non_convex_invest_flow_block.py

```diff
@@ -0,0 +1,639 @@
+        super().__init__(*args, **kwargs)
+
+    def _create(self, group=None):
+        """Creates set, variables, constraints for all flow object with
+        an attribute flow of type class:`.NonConvexInvestFlowBlock`.
+
+        Parameters
+        ----------
+        group : list
+            List of oemof.solph.NonConvexInvestFlowBlock objects for which
+            the constraints are build.
+        """
+        if group is None:
+            return None
+
+        self._create_sets(group)
+        self._create_variables(group)
+        self._create_constraints()
+
+    def _create_sets(self, group):
+        """
+        Nonconvex-related sets similar to the
+        <class 'oemof.solph.flows.NonconvexFlow'> class.
+        """
+        self.MIN_FLOWS = Set(
+            initialize=[(g[0], g[1]) for g in group if g[2].min[0] is not None]
+        )
+        self.STARTUPFLOWS = Set(
+            initialize=[
+                (g[0], g[1])
+                for g in group
+                if g[2].nonconvex.startup_costs[0] is not None
+                or g[2].nonconvex.maximum_startups is not None
+            ]
+        )
+        self.MAXSTARTUPFLOWS = Set(
+            initialize=[
+                (g[0], g[1])
+                for g in group
+                if g[2].nonconvex.maximum_startups is not None
+            ]
+        )
+        self.SHUTDOWNFLOWS = Set(
+            initialize=[
+                (g[0], g[1])
+                for g in group
+                if g[2].nonconvex.shutdown_costs[0] is not None
+                or g[2].nonconvex.maximum_shutdowns is not None
+            ]
+        )
+        self.MAXSHUTDOWNFLOWS = Set(
+            initialize=[
+                (g[0], g[1])
+                for g in group
+                if g[2].nonconvex.maximum_shutdowns is not None
+            ]
+        )
+        self.MINUPTIMEFLOWS = Set(
+            initialize=[
+                (g[0], g[1])
+                for g in group
+                if g[2].nonconvex.minimum_uptime is not None
+            ]
+        )
+
+        self.MINDOWNTIMEFLOWS = Set(
+            initialize=[
+                (g[0], g[1])
+                for g in group
+                if g[2].nonconvex.minimum_downtime is not None
+            ]
+        )
+
+        self.NEGATIVE_GRADIENT_FLOWS = Set(
+            initialize=[
+                (g[0], g[1])
+                for g in group
+                if g[2].nonconvex.negative_gradient["ub"][0] is not None
+            ]
+        )
+
+        self.POSITIVE_GRADIENT_FLOWS = Set(
+            initialize=[
+                (g[0], g[1])
+                for g in group
+                if g[2].nonconvex.positive_gradient["ub"][0] is not None
+            ]
+        )
+
+        # New nonconvex-investment-related set defines in the
+        # <class 'oemof.solph.flows.NonconvexInvestFlow'> class.
+        self.NON_CONVEX_INVEST_FLOWS = Set(
+            initialize=[(g[0], g[1]) for g in group]
+        )
```

Comment:
This name for a `Set` is already used in the `InvestmentFlowBlock`  to have non convex investments (i.e. minimal size that can be installed). If it's not possible to have a non-convex investment when also the flow is non-convex, this has to be docuumented and caught if users still try to do it. In any way, the name has to be different.

Reply:
I [propose to rename](https://github.com/oemof/oemof-solph/pull/856/commits/509d44e6897ebac5896a738c153c4d80905fecc8) the new type of flows to `InvestNonConvexFlow`. The names will hopefully make clear that one is a non-convex investment into a flow while the other is an investment in a non-convex flow.

Reply:
I would understand `InvestNonConvexFlow` as `Invest + NonConvexFlow` and not `NonConvexInvest + Flow`. I would be more in favor of normal flow with non-convex investment being named `NonConvexInvestFlow` and non-convex flow with normal investment being named `InvestmentNonConvexFlow`.

Reply:
@Bachibouzouk : Isn't this is the way it is now in the current PR?

Reply:
You renamed `NonConvexInvestFlow` to `InvestNonConvexFlow` to describe a non-convex investment into a flow, I think the name `NonConvexInvestFlow` is a better name for a non-convex investment into a flow